src/main/kotlin/no/nav/syfo/cronjob/CronjobModule.kt

```diff
@@ -0,0 +1,56 @@
+package no.nav.syfo.cronjob
+
+import kotlinx.coroutines.*
+import no.nav.syfo.application.ApplicationState
+import no.nav.syfo.application.Environment
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.application.mq.MQSender
+import no.nav.syfo.behandler.BehandlerDialogmeldingService
+import no.nav.syfo.cronjob.leaderelection.LeaderPodClient
+import org.slf4j.LoggerFactory
+
+private val log = LoggerFactory.getLogger(DialogmeldingSendCronjob::class.java)
+
+fun cronjobModule(
+    applicationState: ApplicationState,
+    database: DatabaseInterface,
+    environment: Environment,
+    mqSender: MQSender,
+) {
+    val leaderPodClient = LeaderPodClient(
+        environment = environment,
+    )
+    val cronjobRunner = DialogmeldingCronjobRunner(
+        applicationState = applicationState,
+        leaderPodClient = leaderPodClient,
+    )
+    val behandlerDialogmeldingService = BehandlerDialogmeldingService(
+        database = database,
+    )
+    val dialogmeldingSendCronjob = DialogmeldingSendCronjob(
+        behandlerDialogmeldingService = behandlerDialogmeldingService,
+        mqSender = mqSender,
+    )
+
+    createListenerCronjob(
+        applicationState = applicationState,
+    ) {
+        cronjobRunner.start(
+            cronjob = dialogmeldingSendCronjob
+        )
+    }
+}
+
+fun createListenerCronjob(
+    applicationState: ApplicationState,
+    action: suspend CoroutineScope.() -> Unit
+): Job = GlobalScope.launch {
+    try {
+        action()
+    } catch (ex: Exception) {
+        log.error("Something went wrong, terminating application", ex)
+    } finally {
+        applicationState.alive = false
+        applicationState.ready = false
+    }
+}
```

Comment:
Kan gjenbruke `fun launchBackgroundTask()` her.

Reply:
Det er sant. Gjør litt rafaktorering for å få bort den dupliseringen 👍